apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -261,6 +271,7 @@ func validateRootDeviceHints(rdh *RootDeviceHints) error {
 // also make the corresponding changes in the OperationalStatus and
 // ErrorType fields in the struct definition of BareMetalHostStatus in
 // the file baremetalhost_types.go.
+// Deprecated: This method is going to be removed in a next release.
```

Comment:
The `Deprecated` comment should have one empty commented line above to be picked up properly.
On the other hand, I'm not sure we need to deprecate these private functions (they are not capitalized so should not be considered public and usable by anyone outside of this repo).
```suggestion
//
// Deprecated: This method is going to be removed in a next release.
```

Reply:
I was debating when I was first making these changes should the private functions have deprecation notice. But I think it is good to have just in case so contributors don't have to relay on knowledge or memory. And its also a reminder for us to remember and remove these at some point.

Reply:
Ok for me :+1: